src/or/nodelist.c

```diff
@@ -854,6 +865,29 @@ nodelist_assert_ok(void)
   digestmap_free(dm, NULL);
 }
 
+/** Ensure that the nodelist has been created with the most recent consensus.
+ *  If that's not the case, make it so.  */
+void
+ensure_nodelist_freshness(void)
+{
+  /* We don't even have a nodelist: this is a NOP. */
+  if (!the_nodelist) {
+    return;
+  }
+
+  /* We don't even have a recent consensus: this is a NOP */
+  networkstatus_t *ns = networkstatus_get_live_consensus(approx_time());
+  if (!ns) {
```

Comment:
Seems we can't call this without a live consensus. Should we BUG() here?

Reply:
Good point. Done in aa951c72dc.

Reply:
ACK.